### PR TITLE
Add Zip

### PR DESCRIPTION
**Pull request by stephencelis**
This PR adds zip2–6 to Array/Optional.

### CODE REVIEW

**Comment by stephencelis** [comment]
Still needs README docs, but can visit after approval/merge.

File: Sources/Overture/KeyPath.swift [hunk 1]
```diff
@@ -116,13 +116,25 @@ public func mprop<Root, Value>(
...
+//  _ update: @escaping (Value) -> Void
+//  )
+//  -> (Root) -> Void {
+//
+//    return mprop(keyPath)(update)
+//}
+
 public func mver<Root, Value>(
-  _ keyPath: ReferenceWritableKeyPath<Root, Value>,
+  _ keyPath: KeyPath<Root, Value>,
```
Comment (mbrandonw) [inline Sources/Overture/KeyPath.swift]: what's this about?

File: Sources/Overture/Optional.swift [hunk 2]
```diff
@@ -22,3 +22,23 @@ public func map<A, B>(
...
 }
+
+
+/// Transforms a pair of optionals into an optional pair.
+///
+/// - Parameters:
+///   - a: An optional value.
+///   - b: Another optional value.
+/// - Returns: An optional pair of values.
+public func zip<A, B>(_ a: A?, _ b: B?) -> (A, B)? {
```
Comment (mbrandonw) [inline Sources/Overture/Optional.swift]: any reason to not throw this in `ZipOptional.swift`?

Comment (stephencelis) [reply Sources/Overture/Optional.swift]: Just localizing all the higher-order zips in their own file. That way we can just delete them if/when we get variadic generics.

File: Sources/Overture/KeyPath.swift (hunk 1 above)
Comment (stephencelis) [reply Sources/Overture/KeyPath.swift]: oops. was troubleshooting some ambiguity. we need to put these APIs in unison through the paces a bit more.

**Pull request merged**